### DISCUSSION TITLE
Understanding signal strength to fine tune gain setting

### DISCUSSION THREAD

**Discussion by f00b4r0** [szpajder/dumpvdl2#43] [Q&A] [answered]
Hi,

I'm trying to find the sweet spot for setting the receiver gain: looking at the signal stats I see a fair number of signals with *positive* levels (i.e. *above* 0dBFS). Am I right in reading this as overloaded signals? Is my understanding correct that one should aim for the loudest signal to be at or below 0dBFS to optimize the SNR (i.e. keep lowering gain as long as 0 or above signals are seen)?

Thanks!

**Comment by szpajder** [comment, marked as answer]
Yes, exactly.

However lowering the gain to a level where there are absolutely no frames at about 0 dBFS might not necessarily be the sweetest spot, because it will cause many distant weak frames to become lost in the noise. So keep the amount of overloads at a reasonable level (say, 1%, or maybe a few % if you are close to an airport) and you should be fine.

**Reply by f00b4r0** [reply]
Thanks, very clear. Querying the data I log thanks to my little hack shows I'm averaging 20% >= 0dBFS and 1.5% > 0dBFS over any random 24h period, so I assume I'm probably still a bit too hot on the input gain :-)
Cheers!

**Reply by f00b4r0** [reply]
Side suggestion: I haven't looked at the code but it may be possible to add a statsd counter to log loud messages, without having to log the actual signal power. I'll give it a stab if it's feasible. Ideally it would be really useful to have something similar to what dump1090 does, which is logging loud_decoded and loud_undecoded messages (the latter being very useful because you *know* that when they happen, you *are* overloading the receiver).
